core/embed/prodtest/main.c

```diff
@@ -513,9 +603,443 @@ static void test_otp_write_device_variant(const char *args) {
+    vcp_println("ERROR optiga_serialize_metadata error %d for OID 0x%04x.", ret,
+                oid);
+    return false;
+  }
+
+  optiga_set_data_object(oid, true, serialized, size);
+
+  ret =
+      optiga_get_data_object(oid, true, serialized, sizeof(serialized), &size);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_get_metadata error %d for OID 0x%04x.", ret, oid);
+    return false;
+  }
+
+  optiga_metadata metadata_stored = {0};
+  ret = optiga_parse_metadata(serialized, size, &metadata_stored);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_parse_metadata error %d.", ret);
+    return false;
+  }
+
+  if (!optiga_compare_metadata(metadata, &metadata_stored)) {
+    vcp_println("ERROR optiga_compare_metadata failed.");
+    return false;
+  }
+
+  return true;
+}
+
+static bool pair_optiga(void) {
+  // The pairing key may already be written and locked. The success of the
+  // pairing procedure is determined by optiga_sec_chan_handshake(). Therefore
+  // it is OK for some of the intermediate operations to fail.
+
+  // Enable writing the pairing secret to OPTIGA.
+  optiga_metadata metadata = {0};
+  metadata.change = OPTIGA_ACCESS_ALWAYS;
+  set_metadata(OID_KEY_PAIRING, &metadata);  // Ignore result.
+
+  // Generate pairing secret.
+  uint8_t secret[SECRET_OPTIGA_KEY_LEN] = {0};
+  optiga_result ret = optiga_get_random(secret, sizeof(secret));
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_get_random error %d,", ret);
+    return false;
+  }
+
+  // Store pairing secret.
+  ret = optiga_set_data_object(OID_KEY_PAIRING, false, secret, sizeof(secret));
+  /*
+   * TODO: Uncomment. Right now this code will render the device unusable with
+   * unofficial firmware. We need to be able to call AttestationDelete before
+   * this code is enabled.
+   *
+  if (OPTIGA_SUCCESS == ret) {
+    secret_erase();
+    secret_write_header();
+    secret_write(secret, SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN);
+  }
+
+  // Verify whether the secret was stored correctly in flash and OPTIGA.
+  memzero(secret, sizeof(secret));
+  if (secret_read(secret, SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN) !=
+      sectrue) {
+    vcp_println("ERROR Failed to read pairing secret.");
+    return false;
+  }
+  */
+
+  ret = optiga_sec_chan_handshake(secret, sizeof(secret));
+  memzero(secret, sizeof(secret));
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_sec_chan_handshake error %d.", ret);
+    return false;
+  }
+
+  return true;
+}
+
+static void optiga_lock(void) {
+  if (!pair_optiga()) {
+    return;
+  }
+
+  // Delete trust anchor.
+  optiga_result ret =
+      optiga_set_data_object(0xe0e8, false, (const uint8_t *)"\0", 1);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_set_data error %d for 0xe0e8.", ret);
+    return;
+  }
+
+  // Set data object metadata.
+  static const optiga_metadata_item ACCESS_PAIRED = {
+      (const uint8_t *)"\x20\xE1\x40", 3};
```

Comment:
It would be nice if you could make use of the value defined [here](https://github.com/trezor/trezor-firmware/blob/213e9aa2b7ad193102ff4854442c4340b7846783/core/embed/prodtest/main.c#L625C28-L625C28). Perhaps a macro could be useful.

Reply:
7dbd17abb99834baaa3c1866f0a4d908b7090f09